demos/python_demos/clip/client.py

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright (c) 2023 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import sys
+sys.path.append("../../common/python")
+import tritonclient.grpc as grpcclient
+import argparse
+import datetime
+import numpy as np
+from client_utils import print_statistics
+from urllib.request import urlretrieve
+from pathlib import Path
+import os
+from PIL import Image
+
+parser = argparse.ArgumentParser(description='Client for clip example')
+
+parser.add_argument('--url', required=False, default='localhost:9000',
+                    help='Specify url to grpc service. default:localhost:9000')
+parser.add_argument('--input_labels', required=False, default="cat,dog,wolf,tiger,man,horse,frog,tree,house,computer",
+                    help="Specify input_labels to the CLIP model. default:cat,dog,wolf,tiger,man,horse,frog,tree,house,computer")
+parser.add_argument('--image_url', required=False, default='https://storage.openvinotoolkit.org/repositories/openvino_notebooks/data/data/image/coco.jpg',
+                    help='Specify image_url to send to the CLIP model. default:https://storage.openvinotoolkit.org/repositories/openvino_notebooks/data/data/image/coco.jpg')
+parser.add_argument('--iterations', default=1,
+                        help='Number of requests iterations, as default use number of images in numpy memmap. default: 1 ',
+                        dest='iterations', type=int)
+args = vars(parser.parse_args())
+
+iterations = args.get('iterations')
+iteration = 0
+
+client = grpcclient.InferenceServerClient(args['url'])
+
+
+image_url = args['image_url']
+print(f"Using image_url:\n{image_url}\n")
+
+input_name = image_url.split("/")[-1]
+sample_path = Path(os.path.join("data", input_name))
+if not os.path.exists(sample_path):
+    sample_path.parent.mkdir(parents=True, exist_ok=True)
+    urlretrieve(
+        image_url,
+        sample_path,
+    )
+#image = Image.open(sample_path)
+with open(sample_path, "rb") as f:
+    data = f.read()
+
+infer_input = grpcclient.InferInput("image", [len(data)], "UINT8")
+infer_input._raw_content = data
+
+input_labels = args['input_labels'].split(",")
+print(f"Using input_labels:\n{input_labels}\n")
+labels_npy = np.array(input_labels)
+infer_input1 = grpcclient.InferInput("input_labels", [len(labels_npy)], labels_npy.dtype.str)
+infer_input1._raw_content = labels_npy.tobytes()
+
+processing_times = np.zeros((0),int)
```

Comment:
can we use regular python list and convert it to numpy right before passing it to `print_statistics` ? it might be easier to read and numpy does not seem to be necessary